Minesweeper.cpp

```
@@ -4,114 +4,314 @@
 
-	void PlaceBombsOnBoard(TileGrid& board) {
-		const float BOMB_DENSITY = 0.15f; // 15% of tiles are bombs
-		const int numberOfBombs = static_cast<int>(board.size() * board[0].size() * BOMB_DENSITY);
+			for (auto &neighbour : neighbours)
+			{
+				if (neighbour.GetContentOption() == Tile::ContentOption::BOMB) continue;
+				if (!neighbour.IsTileCovered()) continue;
+				
+				IntVector2 coords = neighbour.GetCoords();
+				m_grid[coords.y][coords.x].SetTexture(neighbour.GetContentTextureFilePath());
+				m_grid[coords.y][coords.x].ToggleCovered();
```

cppcheck (2.10-2)

[performance] Function parameter 'sampleEntity' should be passed by const reference.

(passedByValue)